RELEASE.md

```diff
@@ -1,38 +1,321 @@
-# Release
+# Rollkit Release Process
 
-## Release Steps
+This document outlines the release process for all Go packages in the Rollkit repository. The packages must be released in a specific order due to inter-dependencies.
 
-* Update config version in [config/defaults.go](https://github.com/rollkit/rollkit/blob/main/pkg/config/defaults.go)
-* Release new Rollkit version
-* Update [Rollkit/Cosmos-SDK](https://github.com/rollkit/cosmos-sdk) with the newly released Rollkit version
-* Release new Rollkit/Cosmos-SDK version
+## Package Dependency Graph
 
-## Releasing a new Rollkit version
+```ascii
+                        ┌──────────┐
+                        │   core   │ (zero dependencies)
+                        └────┬─────┘
+                             │
+        ┌────────────────────┼────────────────────┐
+        │                    │                    │
+        ▼                    ▼                    ▼
+   ┌─────────┐         ┌─────────┐      ┌──────────────┐
+   │   da    │         │ rollkit │      │execution/evm │
+   └─────────┘         └────┬────┘      └──────────────┘
+                            │
+                ┌───────────┴───────────┐
+                │                       │
+                ▼                       ▼
+      ┌─────────────────┐     ┌─────────────────┐
+      │sequencers/based │     │sequencers/single│
+      └────────┬────────┘     └────────┬────────┘
+               │                       │
+               ▼                       ▼
+      ┌─────────────────┐     ┌─────────────────┐
+      │apps/evm/based   │     │apps/evm/single  │
+      └─────────────────┘     └─────────────────┘
+```
 
-### Two ways
+## Release Order
 
-* [CI and Release Github Actions Workflows](https://github.com/rollkit/rollkit/actions/workflows/ci_release.yml)
-* [Manual](https://github.com/rollkit/rollkit/releases)
+Packages must be released in the following order to ensure dependencies are satisfied:
```

Comment:
We need to make sure there are no replaces. Especially for packages we want to `go install` (such as local-da)